src/main/drivers/compass_ak8963.c

```diff
@@ -99,38 +99,38 @@ typedef enum {
 
 static queuedReadState_t queuedRead = { false, 0, 0};
 
-static bool ak8963SensorRead(uint8_t addr_, uint8_t reg_, uint8_t len_, uint8_t *buf)
+static bool ak8963SensorRead(magDev_t *magDev, uint8_t addr_, uint8_t reg_, uint8_t len_, uint8_t *buf)
 {
-    mpuWriteRegisterI2C(NULL, MPU_RA_I2C_SLV0_ADDR, addr_ | READ_FLAG);   // set I2C slave address for read
-    mpuWriteRegisterI2C(NULL, MPU_RA_I2C_SLV0_REG, reg_);                 // set I2C slave register
-    mpuWriteRegisterI2C(NULL, MPU_RA_I2C_SLV0_CTRL, len_ | 0x80);         // read number of bytes
+    mpu6500SpiWriteRegister(&magDev->bus, MPU_RA_I2C_SLV0_ADDR, addr_ | READ_FLAG);   // set I2C slave address for read
+    mpu6500SpiWriteRegister(&magDev->bus, MPU_RA_I2C_SLV0_REG, reg_);                 // set I2C slave register
+    mpu6500SpiWriteRegister(&magDev->bus, MPU_RA_I2C_SLV0_CTRL, len_ | 0x80);         // read number of bytes
```

Comment:
We need a proper way of accessing gyro read/write functions here. Or differently - re-implement SPI read/writes here specifically for the purpose of accessing AK8963 via the gyro.